1153 update import to handle new pdsfile changes

juzen2003 opened the pull request:
- Fixes #1153 
- Were any Django, import pipeline, table_schema, or dictionary files modified? Y
  - If YES:
    - Database used: re-import on a test database using import_for_tests.sh
    - All Django tests pass: Y
    - FLAKE8 run on modified code: Y
- Were any JavaScript or CSS files modified? N
  - If YES:
    - JSHINT run on all affected files: NA
    - Tested on Chrome, Firefox, Safari, Edge, and iOS: NA
      (Remember to test all browser sizes)
    - Tested for race conditions (with delayed API calls if applicable): NA

Description of changes:
- Update tests results for STIS (wavelength values is changed after re-import)
- Make sure all the opus_type & opus_prioritizer in pdsfile rules (latest master branch in pds-webtools) are in internal_def_product_types.json. The newly added opus_type is "gossi_raw_alternate". Tooltips in detail tab for GOSSI superseded processing files shows up after re-import.
- For NH volset, update to make sure 2001 is imported first and then 1001 after that. Update the tests results for NH after re-import.

Known problems:
NA

juzen2003 committed:
Fixed the code to make sure ./import_for_tests.sh work properly.
- In opus_support.py (line 1950-1968), create get_single_format_function
  and get_single_parse_function to obtain the format/parse info for units
  conversions.
- In do_import.py (line 438-441), replace RANGE_FUNCTIONS with
  get_single_parse_function to get the parse info for unit conversions.
- In obs_mission_hubble.json (line 154-156), fix the import failure by
  updating the value of mult_options for obs_mission_hubble_detector_id.

juzen2003 committed:
Fixed the wavelength values of STIS in API test results due to the new
.LBL files update & re-import.

juzen2003 committed:
Make sure 2001 is imported first and then 1001 second for each New Horizon
volume. That way, the primary filespec will be raw in OPUS (same as pdsfile).
- In import_util.py (line 157-163), if it's NH volset, reverse the childnames
  so that 2001 is imported first and 1001 is imported second.
- In import_for_tests.sh (line 23), add some commented out import for NH,
  will update this later.

juzen2003 committed:
Add new product type "gossi_raw_alternate" to internal_def_product_types.json
so that the product type will have tooltips in detail tab.

juzen2003 committed:
Update the NH import for test database and also update the test results.
- In import_for_tests.sh (line 22), for all NH volumes, import 2001 first
  and then 1001.
- In test_metadata_api.py (line 362-388), update tests to test on NHPELO_1001
  instead of testing on NHPELO_2001 (NHPELO_1001 is the primary filespec).

juzen2003 committed:
Remove unused get_single_format_function from importing opus_support.

juzen2003 committed:
Remove commented out import in do_import.py.

rfrenchseti commented on lib/opus_support.py:
```diff
@@ -1947,7 +1947,25 @@ def parse_form_type(s):
 
-
+def get_single_parse_function(form_type_unit_id):
+    range_func = None
```
We don't use the term "range_func" anymore so this one should be called "parse_func' and the next should be "format_func".

rfrenchseti replied:
Likewise just call it "unit_id" not "form_type_unit_id" to match the rest of this file.

rfrenchseti commented on lib/opus_support.py:
```diff
@@ -1947,7 +1947,25 @@ def parse_form_type(s):
     return form_type, form_type_format, form_type_unit
 
-
+def get_single_parse_function(form_type_unit_id):
```
Add docstring: """Return the parse func for a unit_id with a single non-displayed unit."""

rfrenchseti commented on lib/opus_support.py:
```diff
@@ -1947,7 +1947,25 @@ def parse_form_type(s):
+                                    [2])
+    return range_func
+
+def get_single_format_function(form_type_unit_id):
```
Add docstring: """Return the format func for a unit_id with a single non-displayed unit."""

rfrenchseti commented on opus/import/do_import.py:
```diff
@@ -431,12 +431,10 @@ def update_mult_table(table_name, field_name, table_column, val, label,
-            if range_func_name in opus_support.RANGE_FUNCTIONS:
-                range_func = opus_support.RANGE_FUNCTIONS[range_func_name][1]
+        (form_type, form_type_format,
+         form_type_unit_id) = opus_support.parse_form_type(
```
You removed the `import opus_support` so you definitely shouldn't reference `opus_support` like this. I'm actually surprised this works. I would do what the other files do. See, for example, the top of `apps/search/views.py`. Then remove the `opus_support.` entirely.

rfrenchseti commented on opus/import/do_import.py:
```diff
@@ -431,12 +431,10 @@ def update_mult_table(table_name, field_name, table_column, val, label,
+        (form_type, form_type_format,
+         form_type_unit_id) = opus_support.parse_form_type(
+                            table_column['pi_form_type'])
+        range_func = get_single_parse_function(form_type_unit_id)
```
Again let's not call this a "range_func" but instead a "parse_func".

rfrenchseti reviewed (changes requested):
See if you can find a GOSSI observation that has superseded data to include in test_results_contents. When you run create_test_results.py it will print out the Python code for the associated tests which you can then just copy into test_results_contents.py as the appropriate location.

juzen2003 committed:
Update based on github review feedback, including renaming variables and
adding docstring.
- In opus_support.py (line 1950-1970), rename "range_func" in
  get_single_parse_function and get_single_format_function to parse_func
  and format_func.
- In opus_support.py (line 1951, 1962), add docstring to
  get_single_parse_function and get_single_format_function.
- In do_import.py (line 436), change the way parse_form_type gets called
  because of how opus_support is imported.

juzen2003 committed:
Add new opus id "go_ssi_c0368977800" that has superseded data included in
create_test_results.py, and create tests for it.

juzen2003 committed:
Fixed the indentation at line 370-372 in test_results_contents.py.
